src/Persistence/Sql/Connection.php

```diff
@@ -212,6 +212,16 @@ protected static function connectDbalConnection(array $dsn)
         if (isset($dsn['pdo'])) {
             $pdo = $dsn['pdo'];
         } else {
+            $enforceCharset = [
+                'mysql' => 'utf8mb4',
+                'oci' => 'AL32UTF8',
+            ][$dsn['driverSchema']] ?? null;
+
+            if ($enforceCharset !== null) {
+                $dsn['dsn'] = preg_replace('~; *charset=[^;]+~i', '', $dsn['dsn'])
```

Comment:
So we always enforce utf8? What if user database is not utf8? Previously utf8 was used by default, but user could use different charset too.

Reply:
Because of many hidden issues. UTF-8 is defacto standard. This is about the connection charset, you can store any data you want in binary fields, actually, this is one of the feature thi PR adds and fixes.